src/main/scala/com/databricks/spark/xml/util/ValidatorUtil.scala

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2019 Databricks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.databricks.spark.xml.util
+
+import java.nio.file.Paths
+import javax.xml.validation.{Schema, SchemaFactory}
+import javax.xml.XMLConstants
+
+import com.google.common.cache.{CacheBuilder, CacheLoader}
+
+import org.apache.spark.SparkFiles
+
+/**
+ * Utilities for working with XSD validation.
+ */
+private[xml] object ValidatorUtil {
+
+  // Parsing XSDs may be slow, so cache them by path:
+
+  private val cache = CacheBuilder.newBuilder().softValues().build(
+    new CacheLoader[String, Schema] {
+      override def load(key: String): Schema = {
+        // Handle case where file exists as specified
+        var path = Paths.get(key)
```

Comment:
Maybe we could use Hadoop's Path to allow to access on HDFS too, but I think it can be done separately as well.

Reply:
It's a good point. I had in mind that the common case would be a local file but, hm, maybe not. Hadoop's API would support local files too right?

Reply:
Yeah, it can read local files too. Can be done separately I think. Let me just merge for now.